sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -675,30 +675,32 @@ static void _handle_rtr_adv(gnrc_netif_t *netif, const ipv6_hdr_t *ipv6,
     }
 #endif  /* !CONFIG_GNRC_IPV6_NIB_6LBR */
 #endif  /* CONFIG_GNRC_IPV6_NIB_MULTIHOP_P6C */
-    if (rtr_adv->ltime.u16 != 0) {
-        uint16_t rtr_ltime = byteorder_ntohs(rtr_adv->ltime);
-
-        dr = _nib_drl_add(&ipv6->src, netif->pid);
-        if (dr != NULL) {
-            _evtimer_add(dr, GNRC_IPV6_NIB_RTR_TIMEOUT, &dr->rtr_timeout,
-                         rtr_ltime * MS_PER_SEC);
+    if (!gnrc_netif_is_6lbr(netif)) {
+        if (rtr_adv->ltime.u16 != 0) {
```

Comment:
```suggestion
    if (!gnrc_netif_is_6lbr(netif) && (rtr_adv->ltime.u16 != 0)) {
```

and use the else case for both cases? This would have two advantages: If a downstream router decides to invalidate and the border router has that router for one reason or another in its DRL, this would still work even on the border router and the change would also look less scary ;-).

Reply:
The problem is that the else case does

```C
dr = _nib_drl_get(&ipv6->src, netif->pid);
```

so this would lead to removal of the *default router*

Reply:
> **edit** ah never mind it checks that addr and interface match

;-)